lib/Doctrine/MongoDB/Event/MutableEventArgs.php

```diff
@@ -54,4 +56,32 @@ public function isDataChanged()
     {
         return $this->isDataChanged;
     }
+
+    /**
+     * @since 1.3
+     * @return array
+     */
+    public function getOptions()
+    {
+        return $this->isOptionsChanged ? $this->changedOptions : parent::getOptions();
```

Comment:
@jmikola Looks like this is saying "If I have changed any option at all, only return to me the options I've changed." Seems like if the base object had "animal=dog" and then I add a custom option "color=brown", a call to getOptions would return all of the options (animal=dog,color=brown) instead of just the changed options.  What do you think?